spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/config/EurekaConfigServerBootstrapConfiguration.java

```diff
@@ -109,6 +114,22 @@ public WebClientEurekaHttpClient configDiscoveryWebClientEurekaHttpClient(Eureka
 
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(name = "org.springframework.web.client.RestClient")
+	@ConditionalOnProperty(prefix = "eureka.client", name = "restclient.enabled", havingValue = "true")
```

Comment:
After some more consideration, I actually think we could switch to using `RestClient` by default in case of the non-blocking stack, so let's add `matchIfMissing=true` and `@ConditionalOnProperty(prefix = "eureka.client", name = "restclient.enabled", havingValue = "false")` to the `RestTemplate`-based configuration. Sorry for the confusion.

Reply:
I added related conditions while changing `JerseyClientNotPresentOrNotEnabledCondition` to `RestTemplateEnabledCondition`. Is this what you intended? 🤔

Reply:
<img width="824" alt="스크린샷 2024-10-27 오후 10 53 08" src="https://github.com/user-attachments/assets/5cb5982e-7ec5-4166-8dd2-ead4f43ad990">


I think I understood it a little differently, but I think you should change `JerseyClientNotPresentOrNotEnabledCondition` to `RestTemplateEnabledCondition` for the same issue. Don't you think so??

Reply:
If there is something I misunderstood, please let me know.

1. Modify the `eureka.client.restclient.enabled` annotation I added as follows.

```
@ConditionalOnProperty(prefix = "eureka.client", name = "restclient.enabled", havingValue = "true")
```
->
```
@ConditionalOnProperty(prefix = "eureka.client", name = "restclient.enabled", havingValue = "true", matchIfMissing = true)
```

2. Then, set the following annotation to the configuration beans that are set based on `RestTemplate`.

```
@ConditionalOnProperty(prefix = "eureka.client", name = "restclient.enabled", havingValue = "false")
```

However, if I set it like this, some tests will fail. The reason is that `matchIfMissing` is changed so that it should default to `RestTemplate`, but it changes to `RestClient`, or even `RestTemplate` is not set, so the bean is not configured.